### ISSUE TITLE
Docker Compose & separating processes per container

### ISSUE THREAD

**Issue by iancassidyweb** [magento/magento2devbox-web#12]
At the moment this project creates one big Docker container. Do you have anything on the roadmap to re-jig this using [docker compose](https://docs.docker.com/compose/) ? This would allow you to create multiple containers for the likes of:

- php
- nginx
- solr/eslastic search 
- redis 
- rabbit mq 
- cron 
- varnish 

There are already some good examples of this on GitHub. See the following reference projects: 

`https://github.com/mageinferno/docker-magento2-php `
`https://github.com/fballiano/docker-magento2`
`https://github.com/meanbee/docker-magento2`

**Comment by vrann** [comment]
@iancassidyweb Right now php is married with the apache and php-fpm in one container. 

The rest of the services are separate containers: mysql, rabbitmq, varnish, elasticsearch, etc. Docker compose is used to map all of them together.

**Comment by piotrekkaminski** [comment]
@iancassidyweb some more explanation: Our goal was to create an easy to use development environment, not a production environment. The containers in Magento DevBox should not be used as a model for production use.

Additionally, when we tried separate containers, we encountered stability issues, especially when running multiple sets of containers. Docker is a quickly evolving product but it's not yet sufficiently stable. We are eager to revisit this approach and split the container in components, with option for nginx instead of Apache.
